main.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/structs"
+	"github.com/sirupsen/logrus"
+	"github.com/vshn/appcat-cmd/internal/defaults"
+
+	//"k8s.io/cli-runtime/pkg/printers"
+	"sigs.k8s.io/yaml"
+)
+
+func init() {
+	logrus.SetOutput(os.Stderr)
+}
+
+func main() {
+	code := Main(os.Args, os.Stdin, os.Stdout)
+	os.Exit(code)
+}
+
+func Main(args []string, in io.Reader, out io.Writer) int {
+	plainArgs := args
+	service, err := findServiceType(plainArgs[1])
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	plainArgs, err = cleanInputArguments(plainArgs)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	parameters := parseArgs(plainArgs)
+
+	_, err = decorateType(service, parameters)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	err = writeYAML(service, os.Stdout)
+	if err != nil {
+		logrus.Error(err)
+		return 1
+	}
+
+	return 0
+}
+
+func writeYAML(service interface{}, out io.Writer) error {
+	outYaml, err := yaml.Marshal(service)
+	out.Write(outYaml)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Checks and fixes if input parameters are of type "--foo=bar"
+// returns fixed argument list
+func cleanInputArguments(arguments []string) ([]string, error) {
```

Comment:
This method of argument parsing is a bit wonky. I think it will fail for "--foo" "x=y" (key 'foo' with value 'x=y'), and probably also for "--foo" "--bar" (key 'foo' with value '--bar'). These are of course border cases and I don't mind leaving the code as it is for now (we can always improve things later), but the argument parsing should probably work more like this:
* Single pass over all arguments, have a map of all arguments
* Upon encountering "--(.*)=(.*)" put into map $1 as key and $2 as value
* Upon encountering "--(.*)" put into map $1 as key and next argument (unconditionally) as value (consume two arguments)
* Upon encountering anything else print warning

(the regexes are just for illustration, don't implement something like this with regexes)

Again, I'm OK with as it is for now, but I have to mention this for completeness.

Reply:
Are there use cases with --foo --bar as key/value ? In the event of a missing value we would loose the ability to tell which key is missing it's value, which for longer parameter lists could be a headache.

Reply:
I don't think it is worth discussing this at this point, and arguments can be made in many directions, I just wanted to point out that the current implementation doesn't parse all cases correctly.

Reply:
Alright, I added a fix for the value contains = case and added error for missing values.
Will do a proper refactor if we run into issues with the other edge case in the future.

Reply:
:+1: Thank you!